### PR TITLE
renaming functions to match agreed conventions

### LINKED ISSUE #2413: Evaluate consistency and naming of char vs u32 methods in icu_collections and icu_properties
We inconsistently name methods in the various properties and collections classes that deal with char vs u32. Examples: `contains(char)`, `contains_32(u32)`, `get(char)`, and `get_u32(u32)`, but sometimes it is `get(u32)`. And the `get_u32` name sounds like it is returning a `u32`, similar to `get_ule`, when in fact it is an overload of the `get` method.

Feedback from @markusicu.

Thoughts?

- [ ] @echeran 
- [ ] @Manishearth

### PR DESCRIPTION
**Pull request by dsipasseuth**
fix #2413 

get_ule is still get_ule

### CODE REVIEW

File: components/collections/src/codepointtrie/cptrie.rs [hunk 1]
```diff
@@ -888,11 +888,11 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
     /// assert_eq!(cp, 0x1158E);
     ///
     /// let plane_num: u8 = trie.get(cp);
-    /// assert_eq!(trie.get_u32(cp), plane_num as u32);
+    /// assert_eq!(trie.get32(cp), plane_num as u32);
     /// ```
     // Note: This API method maintains consistency with the corresponding
     // original ICU APIs.
-    pub fn get_u32(&self, code_point: u32) -> u32 {
+    pub fn get32(&self, code_point: u32) -> u32 {
```
Comment (sffc) [inline components/collections/src/codepointtrie/cptrie.rs:895]: Nit: CodePointTrie is weird. In this case, the difference between `get` and `get_u32` is actually the return value; both functions take a u32 argument. I think we should change `get` to `get32`... and name `get_u32` to `get32_u32`?

Comment (Manishearth) [reply components/collections/src/codepointtrie/cptrie.rs:895]: Oh boy that's a weird one. `get32_u32` or `get32_as_u32` seem good

Comment (markusicu) [reply components/collections/src/codepointtrie/cptrie.rs:895]: `get32_u32` would be consistent :-) Adding an infix to make it read better is fine although less predictable.

Comment (sffc) [reply components/collections/src/codepointtrie/cptrie.rs:895]: `get32_u32` would be consistent with `get_ule` (which we're not changing) in the sense that the thing after the `_` is the type of the return value.

**Comment by robertbastian** [comment]
For the failing CI test, you need to update the names in `ffi/diplomat/ffi_coverage/tests/missing_apis.txt`

**Review by robertbastian — DISMISSED** [review]
LGTM but looks like @sffc wants to still review as well

**Review by sffc — COMMENTED** [review]
Please change

- `CodePointTrie::get(u32)` to `::get32(u32)`
- `CodePointTrie::get_ule(u32)` to `::get32_ule(u32)`

and optionally consider adding `CodePointTrie::get(char)`

**Review by sffc — CHANGES_REQUESTED** [review]
.

**Review by sffc — DISMISSED** [review]

**Comment by sffc** [comment]
```
[cargo-make] INFO - Running Task: generated-readme-check
[cargo-make] ERROR - Error while running duckscript: Source: Unknown Line: 18 - 

A README.md file is out-of-sync with lib.rs

If you modified a lib.rs file, please run `cargo make generate-readmes`. If you edited a
README.md file directly, please also update the corresponding lib.rs.

The modified files were:
 M components/collections/codepointtrie_builder/README.md
```

**Review by robertbastian — APPROVED** [review]

File: components/collections/src/codepointinvlist/builder.rs [hunk 2]
```diff
@@ -116,7 +116,7 @@ impl CodePointInversionListBuilder {
     /// let mut builder = CodePointInversionListBuilder::new();
     /// builder.add_u32(0x41);
     /// let check = builder.build();
-    /// assert_eq!(check.contains_u32(0x41), true);
+    /// assert_eq!(check.contains32(0x41), true);
     /// ```
     pub fn add_u32(&mut self, c: u32) {
```
Comment (sffc) [inline components/collections/src/codepointinvlist/builder.rs:121]: Is there a reason we didn't also change `add_u32` to `add32` here?

@robertbastian @dsipasseuth @echeran

Comment (sffc) [reply components/collections/src/codepointinvlist/builder.rs:121]: https://github.com/unicode-org/icu4x/issues/4354

**Pull request merged**